core/app/[locale]/(default)/blog/tag/[tagId]/page.tsx

```diff
@@ -16,19 +17,25 @@ export async function generateMetadata(): Promise<Metadata> {
   return await getBlogMetaData();
 }
 
-async function listBlogPosts(
-  paramsPromise: Promise<{ tagId: string }>,
-  searchParamsPromise: Promise<SearchParams>,
-) {
-  const { tagId } = await paramsPromise;
-  const searchParams = await searchParamsPromise;
+async function listBlogPosts(props: Props) {
+  const { tagId } = await props.params;
+  const searchParams = await props.searchParams;
   const blogPosts = await getBlogPosts({ tagId, ...searchParams });
   const posts = blogPosts?.posts ?? [];
 
   return posts;
 }
 
+async function getPaginationInfo(props: Props) {
+  const { tagId } = await props.params;
+  const searchParams = await props.searchParams;
+  const blogPosts = await getBlogPosts({ tagId, ...searchParams });
+
+  return pageInfoTransformer(blogPosts?.pageInfo);
```

Comment:
🍹 If we do change the `| undefined` in `pageInfoTransformer` we could change this as follows
```suggestion
  return pageInfoTransformer(blogPosts?.pageInfo ?? null);
```

Reply:
@migueloller I think it makes sense to go with `PageInfo | undefined | null`, as it will cover most (if not all) possible cases for the different GraphQL contracts/responses, and prevent the need to use nullish coalescing operators (or other operators) to convert between `undefined` or `null` before calling the function.

Reply:
I think we should have the argument type be `ResultOf<typeof PaginationFragment>` and enforce the data exists before being passed into the function.